src/main/java/com/nokia/uwr/service/turns/TurnHandlerImpl.java

```diff
@@ -31,16 +35,49 @@ public class TurnHandlerImpl implements TurnHandler {
     private final APIClient apiClient;
     private final SignalParserImpl signalParser;
     private final CalculationSignalServiceImpl calculationSignalService;
+    private final BTSParser btsParser;
 
     @Autowired
-    public TurnHandlerImpl(Board board, APIClient apiClient,
-                           SignalParserImpl signalParser, CalculationSignalServiceImpl calculationSignalService) {
+    public TurnHandlerImpl(Board board,
+                           APIClient apiClient,
+                           SignalParserImpl signalParser,
+                           BTSParser btsParser,
+                           CalculationSignalServiceImpl calculationSignalService) {
         this.board = board;
         this.apiClient = apiClient;
         this.signalParser = signalParser;
+        this.btsParser = btsParser;
         this.calculationSignalService = calculationSignalService;
     }
 
+    /**
+     * Main implementation
+     *
+     * @author Piotr Stoklosa
+     */
+    public void initializeCall(List<BTSDescription> descriptionList) {
```

Comment:
Is it somehow connected to "handling turns"? I would think about putting those methods in a new class.